### PR TITLE
Eval mode bugfixes

### PR DESCRIPTION
**Pull request by kylesayrs**
Running image_classification/train.py in eval mode now logs results and writes them to a file called eval.txt

### CODE REVIEW

File: src/sparseml/pytorch/image_classification/utils/helpers.py [hunk 1]
```diff
@@ -397,8 +397,20 @@ def save_model_training(
...
     )
+
     info_path = os.path.join(save_dir, f"{save_name}.txt")
+    write_validation_results(info_path, val_res, epoch=epoch)
+
+    if not save_message_shown:
+        print(f"Saving model for epoch {epoch} to {save_dir} for {save_name}")
+
 
+def write_validation_results(info_path, val_res, epoch=0):
```
Comment (bfineran) [inline src/sparseml/pytorch/image_classification/utils/helpers.py:408]: nit: add param types

File: src/sparseml/pytorch/image_classification/train.py [hunk 2]
```diff
@@ -672,14 +676,21 @@ def train(
...
             max_steps=max_eval_steps,
             baseline_run=True,
         )
-    val_res = None
+
+        LOGGER.info(f"\nInitial validation results: {val_res}")
+
+        eval_results_path = os.path.join(save_dir, "eval.txt")
+        helpers.write_validation_results(eval_results_path, val_res, epoch=0)
+        print(f"Saving initial validation results to {eval_results_path}")
```
Comment (bfineran) [inline src/sparseml/pytorch/image_classification/train.py:692]: nit: let's move this to an info log and move the log to the helper function (remove word initial)

Comment (bfineran) [inline src/sparseml/pytorch/image_classification/train.py:692]: let's move this block into an `if eval_mode` block. no need to save these results for a normal training run

File: src/sparseml/pytorch/image_classification/utils/helpers.py (hunk 1 above)
Comment (bfineran) [inline src/sparseml/pytorch/image_classification/utils/helpers.py:408]: let's make epoch optional and not write it if it's None (no need to write epoch for `eval.txt`)

Comment (bfineran) [inline src/sparseml/pytorch/image_classification/utils/helpers.py:408]: should add `"write_validation_results"` to the `__all__` list

**Review by bfineran — CHANGES_REQUESTED** [review]

**Comment by bfineran** [comment]
also make sure to add a short description of changes to the PR

**Review by bfineran — APPROVED** [review]
LGTM please add a printout of the contents of a sample `eval.txt` before landing

**Review by mgoin — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Comment by kylesayrs** [comment]
c37616ac-a04d-4f05-93f7-d665145834dd_text
My first landing!

**Pull request merged**